### PR TITLE
Add tests for application hooks | Closes #2142

### LINKED ISSUE #2142: Write Tests for Application Hooks
- assets\prototype\application\hooks\useIfMounted.ts
- assets\prototype\application\hooks\useTimeZoneTime.ts

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Adds improvements to `useIfMounted` and `useTimeZoneTime`
- Adds tests for `useIfMounted`
- Adds tests for `useTimeZoneTime`
- Closes #2142

### CODE REVIEW

File: assets/prototype/application/hooks/test/useIfMounted.test.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		const consoleSpy = jest.spyOn(console, 'error').mockImplementation(result.current.increment);
+
+		act(() => {
+			result.current.increment();
+		});
+
+		expect(consoleSpy).toHaveBeenCalled();
+		// value won't have changed
+		expect(result.current.count).toBe(0);
+	});
```
Comment (nicolad) [inline assets/prototype/application/hooks/test/useIfMounted.test.tsx]: This is the main scenario I was looking for, maybe update a bit description with that comment from below:
`it('should throw an error if component is updated without ifMounted after unmount', () => {` ?
or similar wording

Comment (manzoorwanijk) [reply assets/prototype/application/hooks/test/useIfMounted.test.tsx]: Agreed

Comment (manzoorwanijk) [reply assets/prototype/application/hooks/test/useIfMounted.test.tsx]: Done :)

File: assets/prototype/application/services/config/test/data.ts [hunk 2]
```diff
@@ -23,8 +23,8 @@ export const mockEeJsData: JsDataProps = {
 		site_url: '',
 	},
 	default_timezone: {
-		pretty: 'ManZooria',
-		string: 'Manzooropa/ManZooria',
-		offset: +27.333,
+		pretty: 'Calcutta',
+		string: 'Asia/Calcutta',
+		offset: +5.5,
```
Comment (nicolad) [inline assets/prototype/application/services/config/test/data.ts]: You didn't like `ManZooria` or it's not parsing it correctly?

Comment (manzoorwanijk) [reply assets/prototype/application/services/config/test/data.ts]: I would have been eternally there but JS doesn't know (🤷‍♂ ) about me 😎

File: assets/prototype/application/hooks/useTimeZoneTime.ts [hunk 3]
```diff
@@ -9,14 +11,16 @@ interface TimeZoneTime {
 }
 
 const useTimeZoneTime = (): TimeZoneTime => {
-	const { config } = useConfig();
+	const config = useConfig();
+
+	const timezone = path<string>(['config', 'timezone', 'name'], config);
```
Comment (tn3rb) [inline assets/prototype/application/hooks/useTimeZoneTime.ts]: 👍

Comment (nicolad) [reply assets/prototype/application/hooks/useTimeZoneTime.ts]: yes, nice

**Pull request merged**